prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package verifyowners
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/repoowners"
+)
+
+const (
+	pluginName     = "verify-owners"
+	ownersFileName = "OWNERS"
+)
+
+var (
+	invalidOwnersLabel = "do-not-merge/invalid-OWNERS"
```

Comment:
We're going to want to add this to the `tide` configs so they don't merge when this is present. That will also require a change to the PR dashboard, but those two changes can land after this PR is in!